src/api/services/PendingValidationInfoDispatcher.ts

```diff
@@ -0,0 +1,166 @@
+        if (!isText) {
+            return null;
+        }
+        const me = guild?.members.me;
+        const colour = me instanceof GuildMember ? me.displayHexColor : "#0099ff";
+        const avatarUrl = me.displayAvatarURL({size: 1024});
+        const infoEmbed = new EmbedBuilder()
+            .setTitle("New Entry")
+            .setColor(colour)
+            .setFields([
+                {
+                    name: "Wad Name",
+                    value: payload.wadName
+                },
+                {
+                    name: "Email",
+                    value: payload.email
+                }
+            ]).setAuthor({
+                name: me.displayName,
+                iconURL: avatarUrl
+            })
+            .setTimestamp();
+        if (payload.info) {
+            infoEmbed.setDescription(payload.info);
+        }
+        if (payload.submitterName) {
+            infoEmbed.addFields([{
+                name: "Submitter Name",
+                value: payload.submitterName
+            }]);
+        }
+        const verifyButton = new ButtonBuilder()
+            .setLabel("Verify")
+            .setStyle(ButtonStyle.Success)
+            .setCustomId("verify");
+        const deleteButton = new ButtonBuilder()
+            .setLabel("Delete")
+            .setStyle(ButtonStyle.Danger)
+            .setCustomId("delete");
+
+        const buttonRow =
+            new ActionRowBuilder<MessageActionRowComponentBuilder>().addComponents(
+                verifyButton, deleteButton
+            );
+        const message = await channelTOPostTo.send({
+            embeds: [infoEmbed],
+            components: [buttonRow]
+        });
+        const collector = message.createMessageComponentCollector();
+        collector.on("collect", async (collectInteraction: ButtonInteraction) => {
+            try {
+                await collectInteraction.deferUpdate();
+                const buttonId = collectInteraction.customId;
+
+                let ok = false;
+                let messageStr = "";
+                if (buttonId === "delete") {
+                    ok = await this.deleteSubmission(payload.id);
+                    messageStr = "deleted";
+                } else if (buttonId === "verify") {
+                    ok = await this.verifySubmission(payload.id);
+                    messageStr = "verified";
+                }
+
+                if (ok) {
+                    const followUp = await collectInteraction.followUp({
+                        content: `Submission has been ${messageStr} successfully`
+                    });
+                    await message.delete();
+                    setTimeout(() => {
+                        followUp.delete().catch();
+                    }, 3000);
+                } else {
+                    const followup = await collectInteraction.followUp({
+                        content: "Unable to delete"
```

Comment:
or verify, right? maybe
``` content: `Unable to ${buttonId}` ```

Reply:
assuming the ID wull always be the action, yes! i can do that